drone-webdriver/src/main/java/org/jboss/arquillian/drone/webdriver/binary/downloading/source/PhantomJSGitHubBitbucketSource.java

```diff
@@ -20,32 +18,44 @@
 
     private static String BASE_DOWNLOAD_URL = "https://bitbucket.org/ariya/phantomjs/downloads/phantomjs-";
 
+    private static String lastPhantomJSRelease = "2.1.1";
+
     public PhantomJSGitHubBitbucketSource(HttpClient httpClient, GitHubLastUpdateCache gitHubLastUpdateCache) {
         super("ariya", "phantomjs", httpClient, gitHubLastUpdateCache);
     }
 
-    @Override
+    /**
+     * As there was announced the end of the development of PhantomJS:
+     * https://groups.google.com/forum/#!msg/phantomjs/9aI5d-LDuNE/5Z3SMZrqAQAJ
+     * it is not necessary to check the latest release - it is expected that there won't be any newer than the last
+     * one: 2.1.1
+     * If the development of PhantomJS is resurrected, then the logic will be uncommented.
+     */
     public ExternalBinary getLatestRelease() throws Exception {
-
-        final HttpClient.Response response =
-            sentGetRequestWithPagination(getProjectUrl() + TAGS_URL, 1, lastModificationHeader());
-        final ExternalBinary latestPhantomJSBinary;
-
-        if (response.hasPayload()) {
-            JsonArray releaseTags = getGson().fromJson(response.getPayload(), JsonElement.class).getAsJsonArray();
-            if (releaseTags.size() == 0) {
-                return null;
-            }
-            String version = releaseTags.get(0).getAsJsonObject().get(TAG_NAME).getAsString();
-            latestPhantomJSBinary = new ExternalBinary(version);
-
-            latestPhantomJSBinary.setUrl(getUrlForVersion(version));
-            getCache().store(latestPhantomJSBinary, getUniqueKey(), extractModificationDate(response));
-        } else {
-            latestPhantomJSBinary = getCache().load(getUniqueKey(), ExternalBinary.class);
-        }
-
-        return latestPhantomJSBinary;
+        // See the javadoc of this method for explanation why this code is commented out
+
+        //        final HttpClient.Response response =
+        //            sentGetRequestWithPagination(getProjectUrl() + TAGS_URL, 1, lastModificationHeader());
+        //        final ExternalBinary latestPhantomJSBinary;
+        //
+        //        if (response.hasPayload()) {
+        //            JsonArray releaseTags = getGson().fromJson(response.getPayload(), JsonElement.class).getAsJsonArray();
+        //            if (releaseTags.size() == 0) {
+        //                return null;
+        //            }
+        //            String version = releaseTags.get(0).getAsJsonObject().get(TAG_NAME).getAsString();
+        //            latestPhantomJSBinary = new ExternalBinary(version);
+        //
+        //            latestPhantomJSBinary.setUrl(getUrlForVersion(version));
+        //            getCache().store(latestPhantomJSBinary, getUniqueKey(), extractModificationDate(response));
+        //        } else {
+        //            latestPhantomJSBinary = getCache().load(getUniqueKey(), ExternalBinary.class);
+        //        }
```

Comment:
Yeah, I wasn't sure which approach I should choose - I was thinking about both of them...